core-java-8/src/main/java/com/baeldung/datetime/AgeCalculator.java

```diff
@@ -0,0 +1,41 @@
+package com.baeldung.datetime;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Period;
+import java.util.Date;
+
+import org.joda.time.Years;
+
+public class AgeCalculator {
+
+    private AgeCalculator() {
+    }
+
+    public static int calculateAgeWithJava8(java.time.LocalDate birthDate, java.time.LocalDate currentDate) {
+        if (birthDate == null || currentDate == null) {
+            throw new IllegalArgumentException("Please enter a valid date!");
+        }
+        return Period.between(birthDate, currentDate).getYears();
+    }
+
+    public static int calculateAgeWithJodaTime(org.joda.time.LocalDate birthDate, org.joda.time.LocalDate currentDate) {
+        if (birthDate == null || currentDate == null) {
+            throw new IllegalArgumentException("Please enter a valid date!");
+        }
+        Years age = Years.yearsBetween(birthDate, currentDate);
+        return age.getYears();
+    }
+
+    public static int calculateAgeWithJava7(Date birthDate, Date currentDate) {
+        if (birthDate == null || currentDate == null) {
+            throw new IllegalArgumentException("Please enter a valid date!");
+        }
+        DateFormat formatter = new SimpleDateFormat("yyyyMMdd");

```

Comment:
That's a curious approach. Of course, there are probably dozens of ways to skin this cat without a library.

I think it would fail once we get to 10000 AD, but it appears safe until then. :)